plugins/logistics.cpp

```diff
@@ -181,16 +190,7 @@ DFhackCExport command_result plugin_onupdate(color_ostream &out) {
     if (!is_enabled || !Core::getInstance().isMapLoaded() || !World::IsSiteLoaded())
         return CR_OK;
     if (world->frame_counter - cycle_timestamp >= CYCLE_TICKS) {
-        int32_t melt_count = 0, trade_count = 0, dump_count = 0, train_count = 0;
-        do_cycle(out, melt_count, trade_count, dump_count, train_count);
-        if (0 < melt_count)
-            out.print("logistics: marked %d item(s) for melting\n", melt_count);
-        if (0 < trade_count)
-            out.print("logistics: marked %d item(s) for trading\n", trade_count);
-        if (0 < dump_count)
-            out.print("logistics: marked %d item(s) for dumping\n", dump_count);
-        if (0 < train_count)
-            out.print("logistics: marked %d animal(s) for training\n", train_count);
+        logistics_cycle(out);
```

Comment:
the reason this logic is kept separate from `logistics_cycle` is because `logistics_cycle` is called in response to manual triggers. We want this auto-running version to be quiet when it does nothing. we want the manually-called version to be clear and verbose when nothing happens.

Reply:
Ok, I’ll revert that.

It feels like there is a lot of duplicate code in a few places like this one example that could be refactored, I didn’t want to go for a bunch in this PR, but separately, do you believe it worth while to reduce the duplication in this or is there a better longer term spot for these features?

Reply:
For instance, what are your thoughts on leaving the call to `logistics_cycle` in `plugin_onupdate` but making the `quiet` aspect an optional input parameter?
![image](https://github.com/DFHack/dfhack/assets/150195674/2551690a-60e6-4e8c-a127-903fa3163002)